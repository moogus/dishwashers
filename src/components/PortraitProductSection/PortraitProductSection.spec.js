import React from 'react';
import { render } from '@testing-library/react';

import useGetProduct from '../../hooks/product/get-product';

import { PortraitProductSection } from './index';

jest.mock('../../hooks/product/get-product');
jest.mock('@reach/router');

describe('<PortraitProductSection />', () => {
  const renderContainer = ({ code = 'foo' } = {}) => {
    const props = { code };
    return render(<PortraitProductSection {...props} />);
  };

  beforeEach(() => {
    jest.clearAllMocks();
    useGetProduct.mockReturnValue({
      title: 'foo',
      media: {
        altText: 'what the dickens',
        urls: ['foo'],
      },
      price: 'Â£1000',
      details: { productInformation: '<p>hiya</p>' },
      displaySpecialOffer: 'baz',
      additionalServices: { includedServices: ['stuff'] },
      code: '12345',
    });
  });

  test('component renders as expected', () => {
    const { container } = renderContainer();
    expect(container).toMatchSnapshot();
  });
});
