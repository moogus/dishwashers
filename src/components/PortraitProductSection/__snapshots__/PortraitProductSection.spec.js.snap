// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortraitProductSection /> component renders as expected 1`] = `
<div>
  <div
    class="d-none d-sm-block d-md-block d-lg-none"
  >
    <div
      class="sc-pFZIQ hXEXXV row"
    >
      <div
        class="sc-kEjbxe vtTif col"
      >
        <div
          class="row"
        >
          <div
            class="sc-gsTCUz jyviIo col"
          >
            <img
              alt="what the dickens"
              class="sc-bdfBwQ eLsnTq d-flex justify-content-center"
              data-testid="selected-image"
              src="foo"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="d-flex justify-content-center"
              data-testid="image-select-buttons"
            >
              <button
                class="sc-dlfnbm elhZeT"
              >
                •
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-jrAGrp dnvivo row pt1 pb1 border-top border-bottom"
        />
        <div
          class="sc-jSgupP irYxfM row"
        >
          <div
            class="col"
          >
            <p
              class="sc-eCssSg fXEDVA"
            >
              £1000
            </p>
            <p
              class="sc-gKsewC eaLtyJ"
            >
              Claim an extra 3 years guarantee via redemption
            </p>
            <p>
              stuff
            </p>
          </div>
        </div>
        <div
          class="sc-iqHYGH kovbwb row"
        >
          <div
            class="col"
          >
            <h2
              class="sc-iBPRYJ fYRKeW"
            >
              Product Information
            </h2>
          </div>
        </div>
        <div
          class="sc-iqHYGH kovbwb row"
        >
          <div
            class="col"
          >
            <h2
              class="sc-fubCfw cwjpul"
            >
              Product code: 12345
            </h2>
          </div>
        </div>
        <div
          class="sc-iqHYGH kovbwb row"
        >
          <div
            class="col"
          >
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
